### PR TITLE
chore(storage): Adjust the tier_compaction limit to reduce the impact on ci

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

e2e tests found that a large number of compaction task ~affected the batch query latency.~

~It is suspected that the frequent compaction task causes more cache misses, which in turn increases the latency.~

I have observed that cg3 accumulates hundreds of sub_levels in the whole batch e2e test, which will affect our read performance. I have observed that cg3 accumulates hundreds of sub_levels in the whole batch e2e test, which will affect our read performance. Based on this information, I found a factor:
1.  before this pr, the triggering of trivial_move was restricted and when the level_size was small, trivial_move was deferred until pick_whole_level. However, after the introduction of the new picker logic, this practice was wrong in the batch e2e test. The test data of batch e2e is not overlapping, and trivial_move cannot be triggered, which will cause tier_compaction pick_multi_level to not be triggered either (too few selectable levels in key_range). Also, the batch e2e test data is too small to trigger the base compaction, resulting in all the data piling up at the sub_level.

Changes to this pr
1. increase the value of DEFAULT_MIN_SUB_LEVEL_COMPACT_LEVEL_COUNT from 3 to 6
2. increased the value of max_depth

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by lmatz** [comment]
> e2e tests found that a large number of compaction task affected the batch query latency.

if we revert it, does it affect the performance of streaming jobs?

**Review by zwang28 — APPROVED** [review]

**Comment by Li0k** [comment]
> > e2e tests found that a large number of compaction task affected the batch query latency.
> 
> if we revert it, does it affect the performance of streaming jobs?
> 
> If it may lead to large performance impact, I suggest we build two images, one with and one without PR, both based on latest `main` to evaluate the regression.

It might, so I try to make as few changes as possible, we need to pay more attention, The failure rate of e2e on main is so high that I prefer merging first to solve the problem

**Comment by Li0k** [comment]
### e2e test
- before
image

- after #8996 
image

The e2e test shows that after 8996 the e2e time consumption increases and the timeout causes the failure rate to increase.

### reason
I tried to reproduce the problem locally and observe the behavior of the system via grafana.
The number of l0_sub_levels is not as expected, according to grafana, there is only one file per sub_level. The logs show that there is no overlap between them and no trivival_move is triggered

- image

- image

- image


In the current logic, the size of the sst file in trivial_move is limited. However, in the e2e test, the sst has fewer data and hits this condition, so trivial_move cannot be triggered. In the absence of trivial_move and the e2e test data are not overlap, the new picker cannot select enough levels, so it cannot trigger tier_compaction. Due to the small amount of data, base-level compaction cannot be triggered either.

image

TLDR
1. The selected file is too small to trigger trivial_move
2. the data in the batch e2e test is not overlap, the non-overlapping-level picker can not pick enough level
3. the amount of data is too small to trigger base-level compaction

### Result
- before
image

- after

- image
- image
- image



image
